### DISCUSSION TITLE
RuntimeError: b' ISO : An error occurred during recv TCP : Connection timed out'

### DISCUSSION THREAD

**Discussion by vporet** [gijzelaerr/python-snap7#476] [Q&A] [answered]
Hello,

I need to use snap7 on a linux docker, I've crashed my RPI and I don't want to use it again.

I've build and run the docker without problems but after _API.connect("192.168.1.232",0,2)_, I've got : **RuntimeError: b' ISO : An error occurred during recv TCP : Connection timed out'.**

I'm sure that PLC is ok because it's ran before my old RPI crash

I did simply this :

import snap7
API = snap7.client.Client()
API.connect("192.168.1.232",0,2)

and after many second : **_RuntimeError: b' ISO : An error occurred during recv TCP : Connection timed out'._**

I can ping my PLC from the docker.

Thanks for help 

Vincent

**Comment by gijzelaerr** [comment, marked as answer]
Closing as stale. This is a network connectivity issue likely related to Docker networking configuration.
